packages-content-model/roosterjs-content-model-core/lib/publicApi/selection/deleteSelection.ts

```diff
@@ -13,12 +17,14 @@ import type {
  * @param model The model to delete selected content from
  * @param additionalSteps @optional Addition delete steps
  * @param formatContext @optional A context object provided by formatContentModel API
+ * @param isCut @optional True if this is a cut operation, false if this is a delete operation
  * @returns A DeleteSelectionResult object to specify the deletion result
  */
 export function deleteSelection(
     model: ContentModelDocument,
     additionalSteps: (DeleteSelectionStep | null)[] = [],
-    formatContext?: FormatWithContentModelContext
+    formatContext?: FormatWithContentModelContext,
+    isCut?: boolean
```

Comment:
deleteSelection is a very common API and is used in a lot of places, so it should not know anything about copy/cut.

Take one step back, in what case we will delete selection but want to keep an empty list number?

Reply:
When we merge models, when we paste content into a list. With this code, the list will be removed. Maybe we can should add this code after into normalize the model?

Reply:
In that case, you can wrap the delete list step as a DeleteSelectionStep function, and pass to the `additionalSteps` parameter (the 2nd parameter of deleteSelection), no need to introduce a new parameter. Then from the cut code, pass that step in.

Reply:
I mean only when cut, pass in this additional step. For all others, keep no change.